packages/react-internal/src/components/FocusTrapZone/FocusTrapZone.tsx

```diff
@@ -309,9 +313,15 @@ export const FocusTrapZone: React.FunctionComponent<IFocusTrapZoneProps> & {
       if (!disableFirstFocus && !elementContains(root.current, internalState.previouslyFocusedElementOutsideTrapZone)) {
         focus();
       }
+      if (!unmodalize && root.current && enableAriaHiddenSiblings) {
+        unmodalize = modalize(root.current);
+      }
     } else if (!newForceFocusInsideTrap || newDisabled) {
       // Transition from forceFocusInsideTrap / FTZ enabled to disabled.
       returnFocusToInitiator();
+      if (unmodalize) {
```

Comment:
Thanks for adding this part! However I think it needs to be handled a bit differently to replicate the class component behavior: the `unmodalize` callback should be stored in `internalState`, and in addition to this part you'll need to add another useEffect for componentWillUnmount.

I started trying to make this change last night and then noticed potentially a bunch of other issues with lifecycles that may not have been ported properly...so we definitely want to test this one carefully.

Reply:
I think I did this properly but let me know which other issues you've found.